exchange/utils.go

```diff
@@ -49,6 +49,7 @@ func cleanOpenRTBRequests(ctx context.Context,
 	gdprPermsBuilder gdpr.PermissionsBuilder,
 	tcf2ConfigBuilder gdpr.TCF2ConfigBuilder,
 	hostSChainNode *openrtb2.SupplyChainNode,
+	bidderInfos config.BidderInfos,
```

Comment:
We already have way too many argument. Perhaps we can leave this out for now and address the needs individually. For example, the gpp supported flag might fit nicely in the privacyConfig object.

Reply:
Should we pass the `bidderInfos` parameter all the way down to the `getAuctionBidderRequests(...)` function as part of this PR? It's likely that we'll need it there.
```
 75 
 76      var allBidderRequests []BidderRequest
 77 -    allBidderRequests, errs = getAuctionBidderRequests(auctionReq, requestExt, bidderToSyncerKey, impsByBidder, aliases, hostSChainNode)
    +    allBidderRequests, errs = getAuctionBidderRequests(auctionReq, requestExt, bidderToSyncerKey, impsByBidder, aliases, hostSChainNode, bidderInfos)
 78 
 79     bidderNameToBidderReq := buildBidResponseRequest(req.BidRequest, bidderImpWithBidResp, aliases, auctionReq.BidderImpReplaceImpID)
 80     //this function should be executed after getAuctionBidderRequests
 81     allBidderRequests = mergeBidderRequests(allBidderRequests, bidderNameToBidderReq)
 82
 83 +-- 98 lines: var gpp gpplib.GppContainer-------------------------------------------------------------------------------------
181     return
182 }
183
184 +-- 33 lines: func ccpaEnabled(account *config.Account, privacyConfig config.Privacy, requestType config.ChannelType) bool {--
217
218   func getAuctionBidderRequests(auctionRequest AuctionRequest,
219       requestExt *openrtb_ext.ExtRequest,
220       bidderToSyncerKey map[string]string,
221       impsByBidder map[string][]openrtb2.Imp,
222       aliases map[string]string,
223 -     hostSChainNode *openrtb2.SupplyChainNode) ([]BidderRequest, []error) {
    +     hostSChainNode *openrtb2.SupplyChainNode,
    +     bidderInfos config.BidderInfos) ([]BidderRequest, []error) {
224  
225       bidderRequests := make([]BidderRequest, 0, len(impsByBidder))
226       req := auctionRequest.BidRequestWrapper
227*-- 14 lines: explicitBuyerUIDs, err := extractBuyerUIDs(req.BidRequest.User)--------------------------------------------------
241  
242       var errs []error
243       for bidder, imps := range impsByBidder {
244           coreBidder := resolveBidder(bidder, aliases)
245  
    +         // We'll probably have to pass a 2.5 or 2.6 request copy depending on the value of the bidder's openrtb flag here
246 -         reqCopy := *req.BidRequest
    +         reqCopy := getVersionCompatibleBidRequest(req.BidRequest, coreBidder, bidderInfos)
247           reqCopy.Imp = imps
exchange/utils.go
```

Reply:
Its not a single flag, but a map of bidder names to their info data. So we would need a map[string]bool or something like that, and a code loop on startup to create this new map. It will also be the only piece in the privacy object that is mapped to bidders. I was also thinking it might future-proof the code a bit, depending on what other openrtb feature flags that come up the we might want to handle in `cleanOpenRTBRequests()`. The 2.5<->2.6 transforms I was thinking either belonged in here, or possibly as a wrapper. And there may be other 2.6 compatibility gymnastics we may have to do in the future on a per bidder basis.

TLDR: Could shove the gpp compatibility flag from BidderInfos into the Privacy config struct, which is not a great fit and already exists elsewhere in the general config struct. But there are other pieces of the BidderInfos we may need in `cleanOpenRTBRequests()` that are not in any way privacy configs that we may need now or in the future.

Reply:
Additional thought, while responding to Gus, I considered the utility of making this function a method of the exchange.  This will allow us to drop 6 other fields from the argument list, giving us a great savings in fields being passed. I would just have to look through the code to insure there is nothing written that depends on these being a copy of the exchange fields (created on passing into the function) rather than the original fields from the exchange. I would be surprised if there is something subtle going on here like that.